debian/ubuntu-advantage-tools.postinst

```diff
@@ -302,6 +303,25 @@ notify_wrong_fips_metapackage_on_cloud() {
     fi
 }
 
+enable_periodic_license_check() {
+    cloud_id=$(cloud-id 2>/dev/null) || cloud_id=""
+    if echo "$cloud_id" | grep -q "gce"; then
+        if check_is_lts "${UBUNTU_CODENAME}"; then
+            touch $LICENSE_CHECK_MARKER_FILE
+        else
+            rm $LICENSE_CHECK_MARKER_FILE || true
+        fi
+    fi
+    # TODO remove this next line, just using it for testing
+    # touch $LICENSE_CHECK_MARKER_FILE
+
+    # HEY REVIEWERS LOOK HERE this next line seems scary to have here
+    systemctl restart ua-license-check.timer
+
+    # TODO remove this next line, just using it for testing
+    # systemctl status ua-license-check.timer
```

Comment:
Hrm this is complicated: I had thought we didn't need to call systemctl restart ua-license-check.timer ourselves. We- instead rely on systemd to start the service for us by using a systemd.path and systemd.service file something like this
```
cat > /lib/systemd/system/ua-license-check.path <<EOF
[Unit]
Description=Ubuntu Advantage check for GCP Ubuntu Pro licenses

[Path]
PathExists=/var/lib/ubuntu-advantage/marker-license-check
Unit=ua-license-check.service

[Install]
WantedBy=multi-user.target
EOF

cat > /lib/systemd/system/ua-license-check.service <<EOF
[Unit]
Description=Ubuntu Advantage check for GCP Ubuntu Pro licenses

[Service]
Type=oneshot
ExecStart=/usr/bin/python3 /usr/lib/ubuntu-advantage/gcp_auto_attach.py
TimeoutSec=0

[Install]                 
WantedBy=multi-user.target

EOF
````



But that is a path conditional which would activate the related systemd service for us at the presence of the marker file at any time after boot.   I'm not certain offhand how we fold the systemd timer into that equation, unless maybe we set Unit=ua-license-check.timer instead of Unit=ua-license-check.service in the ua-license-check.path file.

Reply:
Aha I totally got this mixed up. Thanks for this @blackboxsw !!!

I'll play with it to try and get the path file to activate the timer